src/nikonmn_int.cpp

```diff
@@ -2806,12 +2841,251 @@ fmountlens[] = {
         }
+        auto temp = value.toLong();
+        if (temp == 0 || temp == 255)
+            return os << _("n/a");
+
+        std::ostringstream oss;
+        oss.copyfmt(os);
+        os << std::fixed << std::setprecision(2) << temp;
+        os.copyfmt(oss);
+        os.flags(f);
+        return os;
+    }
+
+    std::ostream& Nikon3MakerNote::printExternalFlashData1(std::ostream& os,
+                                                           const Value& value,
+                                                           const ExifData*)
+    {
+        std::ios::fmtflags f( os.flags() );
+        if (value.count() != 1 || value.typeId() != unsignedByte) {
+            os << "(" << value << ")";
+            os.flags(f);
+            return os;
+        }
+        std::ostringstream oss;
+        oss.copyfmt(os);
+        os << (value.toLong() & 0x80 ? _("External flash zoom override") : _("No external flash zoom override"));
+        os << ", ";
+        os << (value.toLong() & 0x01 ? _("external flash attached") : _("external flash not attached"));
+
+        os.copyfmt(oss);
+        os.flags(f);
+        return os;
+    }
+
+    std::ostream& Nikon3MakerNote::printExternalFlashData2(std::ostream& os,
+                                                           const Value& value,
+                                                           const ExifData*)
+    {
+        std::ios::fmtflags f( os.flags() );
+        if (value.count() != 1 || value.typeId() != unsignedByte) {
+            os << "(" << value << ")";
+            os.flags(f);
+            return os;
+        }
+
         std::ostringstream oss;
         oss.copyfmt(os);
-        os << std::fixed << std::setprecision(2) << value.toLong();
+        long temp = value.toLong();
+
+        switch (temp & 0x07) {
+        case 0:
+            os << _("n/a");
+            break;
+        case 1:
+            os << _("Ready");
+            break;
+        case 6:
+            os << _("Not ready");
+            break;
+        default:
+            os << "(" << temp << ")";
+            break;
+        }
+
+        os.copyfmt(oss);
+        os.flags(f);
+        return os;
+    }
+
+    std::ostream& Nikon3MakerNote::printFlashCompensation(std::ostream& os,
+                                                          const Value& value,
+                                                          const ExifData*)
+    {
+        std::ios::fmtflags f( os.flags() );
+        if (value.count() != 1 || value.typeId() != signedByte) {
+            os << "(" << value << ")";
+            os.flags(f);
+            return os;
+        }
+        std::ostringstream oss;
+        oss.copyfmt(os);
+        float temp = ( value.toFloat()/float(-6.0) );
+        temp *= float(1.00001);                 // Avoid round-off errors
+
+        if (temp == 0)
+            os << 0;
+        else if ( std::numeric_limits<float>::min() > temp || temp > std::numeric_limits<float>::max() )
```

Comment:
```suggestion
        else if (!std::isfinite(temp))
```

Reply:
I hadn't used this before but will in future. Thanks.